### ISSUE TITLE
[java]a:false positive in UnusedPrivateMethodRule,when using spring annotation @PostConstruct

### ISSUE THREAD

**Issue by phinehasz** [pmd/pmd#1268]
[java]

**Affects PMD Version:** 

**Rule:UnusedPrivateMethodRule

**Description:When using spring annotation @PostConstruct,on my private method,this rule also addViolation to this method.
I am aware of this rule extends AbstractIgnoredAnnotationRule.**Have PMD provides some other configs to avoid this code case?**
I just modified this rule,and add a method to ignore annotation.
And I find may I can use ignoredAnnotationsDescriptor.
But I still hope this rule can auto ignore these annotations.

**Code Sample demonstrating the issue:**

```
import javax.annotation.PostConstruct;

public class HttpClientManager{	
	public HttpClientManager httpClientManager = new HttpClientManager();	
	public HttpClientManager(){};

	@PostConstruct
	private void init(){//private method
		httpClientManager.setMaxTotal(100);
	}
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by jsotuyod** [comment]
@phinehasz as you correctly pointed out, this rule extends `AbstractIgnoredAnnotationRule`,  and that means, it exposes a property you can configure to have a given annotation ignored.

[As you can find on the documentation](https://pmd.github.io/pmd-6.6.0/pmd_rules_java_bestpractices.html#unusedprivatemethod), the rule has a property called `ignoredAnnotations`, which defaults to `java.lang.Deprecated`. The rule supports multiple values by using `|` as separator.

You can override the rule properties' values in the ruleset xml, for instance, in this case you could have it ignore both `@Deprecated` and `@PostConstruct` by doing:

```xml
<rule ref="category/java/bestpractices.xml/UnusedPrivateMethod">
    <properties>
        <property name="ignoredAnnotations">java.lang.Deprecated|javax.annotation.PostConstruct</property>
    </properties>
</rule>
```

You can read more about setting property values [in the documentation](https://pmd.github.io/pmd-6.6.0/pmd_userdocs_configuring_rules.html#rule-properties)

**Comment by phinehasz** [comment]
Thx very much!
